src/custom-types.ts

```diff
@@ -0,0 +1,96 @@
+import { ReactEditor } from 'slate-react';
+import { BaseEditor } from 'slate';
+import { HistoryEditor } from 'slate-history';
+
+import { PluginsEditor, ListEditor, LinkEditor, ImageEditor } from './plugins';
+
+export type Next = () => unknown;
+
+export const MARKS = ['bold', 'italic', 'code', 'underline'] as const;
+export type Mark = typeof MARKS[number];
+export type Elements =
+  | 'h1'
+  | 'h2'
+  | 'h3'
+  | 'h4'
+  | 'h5'
+  | 'paragraph'
+  | 'ul_list'
+  | 'ol_list'
+  | 'list_item'
+  | 'code_block'
+  | 'link'
+  | 'figure'
+  | 'image'
+  | 'image_caption'
+  | 'quote';
+
+export type CustomText = { text: string; children: [] } & {
```

Comment:
```suggestion
export type CustomText = { text: string } & {
``` 
Slate doesn't recognize them as leaf-nodes if they contain a children-array. Because of this all text nodes are rendered as paragraphs, and crashing the editor in my testing.

Seems to work when I removed all the children-array here and in all the text elements.

Reply:
Nice catch. Don't know how I missed this :upside_down_face: